### PR TITLE
*: bump golangci-lint-action

### PR DESCRIPTION
**Pull request by xenowits**
Bump `golangci-lint-action` to `v3` and `golangci-lint` version to `v1.52.2`.

category: misc
ticket: none

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -74,6 +74,10 @@ linters-settings:
         disabled: true
       - name: cognitive-complexity
         disabled: true
+      - name: comment-spacings
+        disabled: true
+      - name: use-any
```
Comment (xenowits) [inline .golangci.yml:79]: https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#use-any

File: .golangci.yml [hunk 2]
```diff
@@ -74,6 +74,10 @@ linters-settings:
         disabled: true
       - name: cognitive-complexity
         disabled: true
+      - name: comment-spacings
```
Comment (xenowits) [inline .golangci.yml:77]: https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#comment-spacings

Comment (corverroos) [reply .golangci.yml:77]: seems like you need a few pragmas: `go:generate and nolint`?

File: .golangci.yml (hunk 1 above)
Comment (corverroos) [reply .golangci.yml:79]: sounds like a good rule, why can't we enable this?

Comment (xenowits) [reply .golangci.yml:79]: enabled this in latest changes 👍

File: .golangci.yml (hunk 2 above)
Comment (xenowits) [reply .golangci.yml:77]: yeah, these pragmas are needed, but this linter fails in code blocks like https://github.com/ObolNetwork/charon/blob/main/app/errors/errors.go#L32-L47. I think this is a bug, so i chose to disable it
Screenshot 2023-07-24 at 13 50 30

Comment (corverroos) [reply .golangci.yml:77]: ok cool, add a comment that it doesn't support the latest go spec comments when disabling it.

**Review by corverroos — APPROVED** [review]

**Pull request merged**